src/main/java/com/desiato/puresynth/services/AuthenticationService.java

```diff
@@ -0,0 +1,82 @@
+package com.desiato.puresynth.services;
+
+import com.desiato.puresynth.models.Session;
+import com.desiato.puresynth.models.User;
+import com.desiato.puresynth.repositories.SessionRepository;
+import com.desiato.puresynth.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.time.LocalDateTime;
+
+
+@Service
+public class AuthenticationService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private SessionRepository sessionRepository;
+
+    private static final Logger logger = LoggerFactory.getLogger(AuthenticationService.class);
+
+    public Optional<String> authenticate(String email, String password) {
+        logger.info("Attempting to authenticate user with email: {}", email);
+
+        Optional<User> optionalUser = Optional.ofNullable(userRepository.findByEmail(email));
+
+        if (optionalUser.isEmpty()) {
```

Comment:
It would be easier to read if you inverted the condition:
```
if (userExists(...) && passwordMatches(...) {
  // save session here
} else {
  return Optional.empty();
}
```

Reply:
* `<Entityname>RequestDTO` for request objects